### ISSUE TITLE
VolgaCTF 2021 Qualifier - Summary

### ISSUE THREAD

**Issue by aszx87410** [aszx87410/ctf-writeups#32]
## Writeups

1. [VolgaCTF 2021 Qualifier - Online Wallet (Part 2)](https://github.com/aszx87410/ctf-writeups/issues/27) 14 solved
2. [VolgaCTF 2021 Qualifier - Static Site](https://github.com/aszx87410/ctf-writeups/issues/28) 38 solved
3. [VolgaCTF 2021 Qualifier - Unicorn Networks](https://github.com/aszx87410/ctf-writeups/issues/29) 48 solved
4. [VolgaCTF 2021 Qualifier - JWT](https://github.com/aszx87410/ctf-writeups/issues/30) 131 solved
5. [VolgaCTF 2021 Qualifier - flask-admin](https://github.com/aszx87410/ctf-writeups/issues/31) 81 solved

**Comment by aszx87410** [comment]
VolgaCTF 2021 Qualifier, [Mar 28, 2021 at 10:42:45 PM]:
intended solution

JSON.parse('{"money":100000000000000000000000000000000000000000000000000000000000001E-60}')

100

select json_extract('{"money":100000000000000000000000000000000000000000000000000000000000001E-60}','$.money')

100.00000000000001

**Comment by PeGrina** [comment]
By the way, it was possible to get a negative value, which also gave the flag.

**Comment by aszx87410** [comment]
Hi @PeGrina, could you elaborate on how to get negative value? I am curious!

**Comment by PeGrina** [comment]
Yes, @aszx87410, I did it in a slightly strange way, I just quickly sent transfer requests, because of this, the server processed 2 parallel requests at once, which is why it took money from my account 2 times. Because of this, out of 100 coins, I have -98 coins left.

**Comment by aszx87410** [comment]
@PeGrina It sounds like race condition? Actually I also solved this one by race condition but I still don't know why it works. There are transaction and `select... for update`, I thought the balance of the wallet will be locked and shouldn't have race condition.
